packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,150 @@
+            ? /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d):(?:[0-5]\d)$/
+            : /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d)$/;
+        }
+        if (!regexString.test(userInput)) {
+          const useHour12ErrorMessage = useHour12 ? '12-hour' : '24-hour';
+          showSeconds
+            ? (errorMessageToDisplay =
+                `TimePicker format must be valid and in the ${useHour12ErrorMessage} ` + `format hh:mm:ss A.`)
+            : (errorMessageToDisplay =
+                `TimePicker format must be valid and in the ${useHour12ErrorMessage} ` + `format hh:mm A.`);
+        }
+        return errorMessageToDisplay;
+      };
+
+      const key = option?.key;
+      let updatedUserText = '';
+      let errorMessageToDisplay = '';
+      if (value) {
+        if (allowFreeform && !option) {
+          if (!onFormatDate) {
+            // Validate only if user did not add onFormatDate
+            errorMessageToDisplay = validateUserInput(value);
+          } else {
+            // Use user provided validation if onFormatDate is provided
+            if (onValidateUserInput) {
+              errorMessageToDisplay = onValidateUserInput(value);
+            }
+          }
+        }
+        updatedUserText = value;
+      } else if (option) {
+        updatedUserText = option.text;
+      }
+
+      setErrorMessage(errorMessageToDisplay);
+      setUserText(updatedUserText);
+      setSelectedKey(key);
+    },
+    [allowFreeform, onFormatDate, onValidateUserInput, showSeconds, useHour12],
+  );
+
+  const evaluatePressedKey = (event: React.KeyboardEvent<IComboBox>) => {
+    if (
+      !onFormatDate &&
+      // Only permit input of digits, space, colon, A/P/M characters
+      !(
+        (event.charCode >= KEYCHAR_0 && event.charCode <= KEYCHAR_COLON) ||
+        event.charCode === KEYCHAR_SPACE ||
+        event.charCode === KEYCHAR_A ||
+        event.charCode === KEYCHAR_M ||
+        event.charCode === KEYCHAR_P
+      )
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <div>
+      <ComboBox
+        allowFreeform={allowFreeform}
+        selectedKey={selectedKey}
+        label={label}
+        errorMessage={errorMessage}
+        options={timePickerOptions}
+        onChange={onChange}
+        text={userText}
+        //eslint-disable-next-line
+        onKeyPress={evaluatePressedKey}
+        {...rest}
+      />
+    </div>
+  );
+};
+
+const generateDefaultTime = (increments: number, timeRange: ITimeRange) => {
+  const defaultTime = new Date();
+  if (timeRange.start >= 0) {
+    defaultTime.setHours(timeRange.start);
+  }
+  const ceiledMinute = ceilMinuteToIncrement(defaultTime.getMinutes(), increments);
+  if (ceiledMinute) {
+    defaultTime.setMinutes(ceiledMinute);
+  }
+  return defaultTime;
+};
+
+const getDropdownOptionsCount = (increments: number, timeRange: ITimeRange) => {
+  let hoursInRange = TimeConstants.HoursInOneDay;
+  if (timeRange.start >= 0 && timeRange.end >= 0) {
```

Comment:
You should probably clamp the start and end to the valid range (0-23) before the calculations

Reply:
Resolved by clamping timeRange using helper function `clampTimeRange`.